File: Binary Tree/100. Same Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:

        def dfs(one, two):
            if not one and not two:
                return True
            if one and two and one.val == two.val:
                return dfs(one.left, two.left) and dfs(one.right,two.right)
            else:
                return False 

        return dfs(p,q)

File: Binary Tree/100. Same Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool isSameTree(struct TreeNode* p, struct TreeNode* q){
    bool dfs(struct TreeNode* _p, struct TreeNode* _q){
        if (!_p && !_q){
            return true;
        }
        else if (_p && _q && _p->val == _q->val){
            return dfs(_p->left, _q->left) && dfs(_p->right, _q->right);
        }
        else {
            return false;
        }        
    }
    return dfs(p, q);
}
